starlette_admin/contrib/beanie/view.py

```diff
@@ -0,0 +1,343 @@
+import functools
+from typing import (
+    Any,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+    get_args,
+    get_origin,
+)
+
+import bson.errors
+import starlette_admin.fields as sa
+from beanie import Document, Link, PydanticObjectId
+from mongoengine.queryset import QNode
+from mongoengine.queryset.visitor import QCombination
+from pydantic import ValidationError
+from starlette.requests import Request
+from starlette_admin._types import RequestAction
+from starlette_admin.contrib.beanie.converters import (
+    BeanieModelConverter,
+)
+from starlette_admin.contrib.beanie.helpers import (
+    Q,
+    build_order_clauses,
+    flatten_qcombination,
+    is_link_type,
+    is_list_of_links_type,
+    resolve_deep_query,
+)
+from starlette_admin.fields import (
+    HasOne,
+    RelationField,
+)
+from starlette_admin.helpers import (
+    not_none,
+    prettify_class_name,
+    pydantic_error_to_form_validation_errors,
+    slugify_class_name,
+)
+from starlette_admin.views import BaseModelView
+
+
+class ModelView(BaseModelView):
+    def __init__(
+        self,
+        document: Type[Document],
+        icon: Optional[str] = None,
+        name: Optional[str] = None,
+        label: Optional[str] = None,
+        identity: Optional[str] = None,
+        converter: Optional[BeanieModelConverter] = None,
+        exclude_fields_from_create: Optional[List[str]] = None,
+        exclude_fields_from_edit: Optional[List[str]] = None,
+        exclude_fields_from_list: Optional[List[str]] = None,
+        exclude_fields_from_detail: Optional[List[str]] = None,
+        full_text_override_order_by: bool = False,
```

Comment:
Let's remove them from here and keep them as class parameters to be consistent with other backends.

Reply:
Done 7083157